Mathlib/Topology/Algebra/Algebra.lean

```diff
@@ -66,14 +70,451 @@ theorem algebraMapCLM_coe : ⇑(algebraMapCLM R A) = algebraMap R A :=
+theorem coe_eq_id {f : A →A[R] A} :
+    (f : A →ₐ[R] A) = AlgHom.id R A ↔ f = ContinuousAlgHom.id R A:= by
+  rw [← coe_id, toAlgHom_inj]
+
+@[simp]
+theorem one_apply (x : A) : (1 : A →A[R] A) x = x := rfl
+
+end id
+
+section comp
+
+variable {R} {A}
+variable {B : Type*} [Semiring B] [TopologicalSpace B] [Algebra R A] [Algebra R B]
+  {C : Type*} [Semiring C] [Algebra R C] [TopologicalSpace C]
+
+/-- Composition of continuous algebra homomorphisms. -/
+def comp (g : B →A[R] C) (f : A →A[R] B) : A →A[R] C :=
+  ⟨(g : B →ₐ[R] C).comp (f : A →ₐ[R] B), g.2.comp f.2⟩
+
+@[simp, norm_cast]
+theorem coe_comp (h : B →A[R] C) (f : A →A[R] B) :
+    (h.comp f : A →ₐ[R] C) = (h : B →ₐ[R] C).comp (f : A →ₐ[R] B) := rfl
+
+@[simp, norm_cast]
+theorem coe_comp' (h : B →A[R] C) (f : A →A[R] B) : ⇑(h.comp f) = h ∘ f := rfl
+
+theorem comp_apply (g : B →A[R] C) (f : A →A[R] B) (x : A) : (g.comp f) x = g (f x) := rfl
+
+@[simp]
+theorem comp_id (f : A →A[R] B) : f.comp (ContinuousAlgHom.id R A) = f :=
+  ext fun _x => rfl
+
+@[simp]
+theorem id_comp (f : A →A[R] B) : (ContinuousAlgHom.id R B).comp f = f :=
+  ext fun _x => rfl
+
+theorem comp_assoc {D : Type*} [Semiring D] [Algebra R D] [TopologicalSpace D] (h : C →A[R] D)
+    (g : B →A[R] C) (f : A →A[R] B) : (h.comp g).comp f = h.comp (g.comp f) :=
+  rfl
+
+instance : Mul (A →A[R] A) := ⟨comp⟩
+
+theorem mul_def (f g : A →A[R] A) : f * g = f.comp g := rfl
+
+@[simp]
+theorem coe_mul (f g : A →A[R] A) : ⇑(f * g) = f ∘ g := rfl
+
+theorem mul_apply (f g : A →A[R] A) (x : A) : (f * g) x = f (g x) := rfl
+
+instance : Monoid (A →A[R] A) where
+  mul_one _ := ext fun _ => rfl
+  one_mul _ := ext fun _ => rfl
+  mul_assoc _ _ _ := ext fun _ => rfl
+
+theorem coe_pow (f : A →A[R] A) (n : ℕ) : ⇑(f ^ n) = f^[n] :=
+  hom_coe_pow _ rfl (fun _ _ ↦ rfl) _ _
+
+/-- coercion from `ContinuousAlgHom` to `AlgHom` as a `RingHom`. -/
+@[simps]
+def toAlgHomMonoidHom : (A →A[R] A) →* A →ₐ[R] A where
+  toFun        := (↑)
+  map_one'     := rfl
+  map_mul' _ _ := rfl
+
+end comp
+
+section prod
+
+variable {R} {A}
+variable {B : Type*} [Semiring B] [TopologicalSpace B] [Algebra R A] [Algebra R B]
+  {C : Type*} [Semiring C] [Algebra R C] [TopologicalSpace C]
+
+/-- The cartesian product of two continuous algebra morphisms as a continuous algebra morphism. -/
+protected def prod (f₁ : A →A[R] B) (f₂ : A →A[R] C) :
+    A →A[R] B × C :=
+  ⟨(f₁ : A →ₐ[R] B).prod f₂, f₁.2.prod_mk f₂.2⟩
+
+@[simp, norm_cast]
+theorem coe_prod (f₁ : A →A[R] B) (f₂ : A →A[R] C) :
+    (f₁.prod f₂ : A →ₐ[R] B × C) = AlgHom.prod f₁ f₂ :=
+  rfl
+
+@[simp, norm_cast]
+theorem prod_apply (f₁ : A →A[R] B) (f₂ : A →A[R] C) (x : A) :
+    f₁.prod f₂ x = (f₁ x, f₂ x) :=
+  rfl
+
+variable {F : Type*}
+
+instance completeSpace_eqLocus {D : Type*} [UniformSpace D] [CompleteSpace D]
+    [Semiring D] [Algebra R D] [T2Space B]
+    [FunLike F D B] [AlgHomClass F R D B] [ContinuousMapClass F D B]
+    (f g : F) : CompleteSpace (LinearMap.eqLocus f g) :=
```

Comment:
Why not `AlgHom.eqLocus`?

Reply:
because it does not exist…

Reply:
Do you believe I should add it (that is, the equivalent for `Subalgebra` of the file  `Mathlib.Algebra.Module.Submodule.EqLocus`?)